src/main/kotlin/org/jetbrains/research/tasktracker/ui/main/panel/MainPluginWindow.kt

```diff
@@ -1,26 +1,37 @@
 package org.jetbrains.research.tasktracker.ui.main.panel
 
+import com.intellij.ide.ui.UISettingsListener
+import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.util.Disposer
 import com.intellij.ui.jcef.JBCefBrowser
+import org.jetbrains.research.tasktracker.ui.main.panel.template.ErrorPageTemplate
+import org.jetbrains.research.tasktracker.ui.main.panel.template.HtmlTemplate
+import org.jetbrains.research.tasktracker.ui.main.panel.template.IndexPageTemplate
 import javax.swing.JComponent
 
 class MainPluginWindow(service: MainWindowService) {
     private val windowBrowser: JBCefBrowser
+    private var currentTemplate: HtmlTemplate
     val jComponent: JComponent
         get() = windowBrowser.component
 
     init {
+        currentTemplate = IndexPageTemplate
         windowBrowser = JBCefBrowser()
-        windowBrowser.setErrorPage { _, _, _ -> // TODO insert parameters to error page
-            defaultPageContent("error")
+        windowBrowser.setErrorPage { errorCode, errorText, failedUrl ->
+            ErrorPageTemplate.pageContent(errorCode.code.toString(), errorText, failedUrl)
         }
-        loadDefaultPage("index")
+        loadDefaultPage(currentTemplate)
+        val app = ApplicationManager.getApplication().messageBus
+        app.connect().subscribe(
+            UISettingsListener.TOPIC,
```

Comment:
you need to use `LafManagerListener`, it will handle all changes. But in this case please fix templates, since for now it does not work for some of them
<img width="1401" alt="Screenshot 2023-07-21 at 13 15 26" src="https://github.com/JetBrains-Research/tasktracker-3/assets/8993915/49e5d64d-ca87-4bf8-a824-5ccc34723fa1">

Reply:
<img width="1427" alt="Screenshot 2023-07-21 at 13 15 35" src="https://github.com/JetBrains-Research/tasktracker-3/assets/8993915/fb183c9b-f13e-4879-abf6-038663ddbef5">